src/mqt/problemsolver/partialcompiler/evaluator.py

```diff
@@ -0,0 +1,75 @@
+from time import time
+from typing import TypedDict
+
+from mqt.problemsolver.partialcompiler.qaoa import QAOA
+
+
+# @dataclass
+class Result(TypedDict):
```

Comment:
A data class would probably be better suited here.

Reply:
I found it easier to work with a `TypedDict` in this case because the keys and values are more easily extractable when creating the result `csv` file.

Reply:
Fair. Might cost some performance though. Not the most critical part of the code, so still fine. Feel free to resolve